webapp/channels/src/utils/markdown/apply_markdown.ts

```diff
@@ -155,6 +155,16 @@ const applyOlMarkdown = ({selectionEnd, selectionStart, message}: ApplySpecificM
         newEnd = Math.max(selectionEnd - (delimiterLength * count), 0);
     } else {
         let count = 0;
+
+        const getLastNumberedLine = (text: string): number => {
+            const match = text.match(/(\d+)\.\s/g);
```

Comment:
I think this regex isn't quite strict enough because there's a few places where it'll match incorrect things:
1. If I type a number at the end of a sentence like
    ```
    We need 12.
    
    This is one of them.
    ```
    If we try to turn "This is one of them" into a list, it'll become "13. This is one of them"
2. Similarly, it can also match numbers which aren't part of the current list.
    ```
    These are fruits:
    1. Apple
    2. Banana
    
    These are animals:
    Dog
    ```
    In this case, "Dog" will become "3. Dog"